### PR TITLE
Nisha | Test | Table headers are getting remove when searching any random value

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/86cv2xwyg

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts [hunk 1]
```diff
@@ -137,39 +137,19 @@ export class ListBranchTransferComponent implements OnInit {
...
-            (hasResponse && this.inlineSearch !== 'senderReceiver' || this.showClearFilter) ||
-            (this.inlineSearch === 'senderReceiver' && !hasResponse || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch === 'senderReceiver' || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch !== 'fromWarehouse' || this.showClearFilter) ||
-            (this.inlineSearch === 'fromWarehouse' && !hasResponse || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch === 'fromWarehouse' || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch !== 'toWarehouse' || this.showClearFilter) ||
-            (this.inlineSearch === 'toWarehouse' && !hasResponse || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch === 'toWarehouse' || this.showClearFilter)
+        const shouldShow = (
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts]:
```suggestion
        return (
```

File: apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts [hunk 2]
```diff
@@ -137,39 +137,19 @@ export class ListBranchTransferComponent implements OnInit {
     public showReceiver = false;
     /** True if translations loaded */
     public translationLoaded: boolean = false;
+    /* True if show header */
+    public showData: boolean = true;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts]: remove variable

File: apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts [hunk 3]
```diff
@@ -137,39 +137,19 @@ export class ListBranchTransferComponent implements OnInit {
...
-            (hasResponse && this.inlineSearch === 'fromWarehouse' || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch !== 'toWarehouse' || this.showClearFilter) ||
-            (this.inlineSearch === 'toWarehouse' && !hasResponse || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch === 'toWarehouse' || this.showClearFilter)
+        const shouldShow = (
+            this.branchTransferForm?.controls['sender']?.value ||
+            this.branchTransferForm?.controls['receiver']?.value ||
+            this.branchTransferForm?.controls['senderReceiver']?.value ||
+            this.branchTransferForm?.controls['fromWarehouse']?.value ||
+            this.branchTransferForm?.controls['toWarehouse']?.value
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts]: ) &&
                    (!this.branchTransferForm?.controls['voucherType']?.value ||
                    !this.branchTransferForm?.controls['amountOperator']?.value ||
                    !this.branchTransferForm?.controls['amount']?.value)

File: apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts [hunk 4]
```diff
@@ -137,39 +137,19 @@ export class ListBranchTransferComponent implements OnInit {
...
-            (this.inlineSearch === 'senderReceiver' && !hasResponse || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch === 'senderReceiver' || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch !== 'fromWarehouse' || this.showClearFilter) ||
-            (this.inlineSearch === 'fromWarehouse' && !hasResponse || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch === 'fromWarehouse' || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch !== 'toWarehouse' || this.showClearFilter) ||
-            (this.inlineSearch === 'toWarehouse' && !hasResponse || this.showClearFilter) ||
-            (hasResponse && this.inlineSearch === 'toWarehouse' || this.showClearFilter)
+        const shouldShow = (
+            this.branchTransferForm?.controls['sender']?.value ||
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts]:
```suggestion
            (this.branchTransferForm?.controls['sender']?.value ||
```

File: apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.html [hunk 5]
```diff
@@ -62,7 +62,7 @@
                 </form>
             </div>
         </div>
-        <section *ngIf="shouldShowElement">
+        <section *ngIf="showData">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.html]:
```suggestion
        <section *ngIf="branchTransferResponse?.length || shouldShowElement">
```

File: apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts [hunk 6]
```diff
@@ -353,9 +333,25 @@ export class ListBranchTransferComponent implements OnInit {
                 this.branchTransferPaginationObject.totalItems = response.body.totalItems;
                 this.branchTransferPaginationObject.count = response.body.count;
                 this.branchTransferResponse = response.body?.items;
+                if(
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/branch-transfer/list-branch-transfer/list-branch-transfer.component.ts]: remove this code

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**